src/components/FormComponents/TextPicker.js

```diff
@@ -23,6 +25,7 @@ export default function TextPicker(props) {
         }}
         placeholder={props.placeholder}
         className={`picker picker--textInput ${props.className}`}
+        {...(testId ? { "data-testid": testId } : {})}
```

Comment:
Added `data-testid` here because it was much better to test that way. Getting component by text was returning multiple elements.